### ISSUE TITLE
collect-totals api call format is incorrect

### ISSUE THREAD

**Issue by michhy** [DivanteLtd/vue-storefront-api#228]
in src/api/cart.js, the api call for collect-totals should be 'PUT' instead of 'POST', 
and the body format is incorrect now.

See reference: https://devdocs.magento.com/redoc/2.3/admin-rest-api.html#operation/quoteCartTotalManagementV1CollectTotalsPut

`cartApi.post('/collect-totals', (req, res) => {
		const cartProxy = _getProxy(req)
		res.setHeader('Cache-Control', 'no-cache, no-store');
		if (!req.body.methods) {
			return apiStatus(res, 'No shipping and payment methods element provided within the request body', 500)
		}
		cartProxy.collectTotals(req.query.token, req.query.cartId ? req.query.cartId : null, req.body.methods).then((result) => {`

should be 
`cartApi.put('/collect-totals', (req, res) => {
		const cartProxy = _getProxy(req)
		res.setHeader('Cache-Control', 'no-cache, no-store');
		if (!req.body.methods) {
			return apiStatus(res, 'No shipping and payment methods element provided within the request body', 500)
		}
		cartProxy.collectTotals(req.query.token, req.query.cartId ? req.query.cartId : null, req.body).then((result) => {`

and when developer call from action.ts in VSF, the payload body should follow the format in magento API reference.

**Comment by pkarw** [comment]
You mean the call format has changed in magento 2.3.x? Because it was surely fine with 2.2.
In such a case we need to add a check in vs-api platform adapter and support both formats by adapting the new result to old format. We can’t modify the data format that vs front is using

**Comment by pkarw** [comment]
By the way not sure if collect-totals is used anywhere in the front so as we’re rather using /totals

**Comment by michhy** [comment]
Yes as I have activated the cart rule with payment method as condition in magento, I need to display that discount in cart summary, and thus I found out only collect-totals accepts paymentMethod code changes to supply updated discount. Anyways this collect cart token is not used in default VSF.

**Comment by pkarw** [comment]
Oh I see - pretty interesting case. Could you propose a PR with a general change? Otherwise you might want to use magento2-rest-client addMethods to add this collect totals: https://github.com/DivanteLtd/magento2-rest-client

**Comment by michhy** [comment]
OK sure
